dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClasspathChecker.java

```diff
@@ -101,21 +94,12 @@ public static void main(String[] arguments)
       throws IOException, RepositoryException, ParseException {
 
     CommandLine commandLine = ClasspathCheckOption.readCommandLine(arguments);
-    ImmutableList<Path> inputClasspath = ClasspathCheckOption.generateInputClasspath(commandLine);
-
-    ImmutableSet.Builder<Path> entryPoints = ImmutableSet.builder();
-    if (commandLine.hasOption("a") || commandLine.hasOption('b')) {
-      // For an artifact list (or a BOM), the first elements in inputClasspath are the artifacts
-      // specified the list, followed by their dependencies.
-      int artifactCount = ClasspathCheckOption.generateArtifacts(commandLine).size();
-      // For Maven artifact list (or a BOM), entry point classes are ones in the list
-      entryPoints.addAll(inputClasspath.subList(0, artifactCount));
-    } else {
-      // For list of jar files, entry point classes are all classes in the files
-      entryPoints.addAll(inputClasspath);
-    }
+    ImmutableList<Path> inputClasspath = ClasspathCheckOption.parseInputClasspath(commandLine);
+
+    ImmutableSet<Path> entryPointJars =
+        ClasspathCheckOption.parseEntryPointJars(commandLine, inputClasspath);
```

Comment:
For later, it might be nice for `ClasspathCheckOption` to hold info about the command line in fields instead of having static methods. Then this method could look more like:

```java
Arguments arguments = Arguments.readCommandLine(arguments);
ClasspathChecker classpathChecker = create(arguments.inputClassPath(), arguments.entryPointJars());
```

Reply:
Created an issue #419